contracts/lootbox-v1/contracts/Lootbox.sol

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./interfaces/IUniswapV2Router02.sol";
+import "./interfaces/IUbiquityAlgorithmicDollar.sol";
+import "./interfaces/IWyvernExchange.sol";
+
+contract Lootbox is Ownable {
+    IUbiquityAlgorithmicDollar public uADToken;
+    address public uCRToken;
+
+    uint256 public transactionFee;
+    uint256 public cashbackPercentage;
+    address private constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+
+    IUniswapV2Router02 public uniswapRouter;
+
+    modifier exceptBeforeBuy(uint256 _amountuAD) {
+        require(
+            IERC20(uADToken).balanceOf(msg.sender) >= _amountuAD * 10**18,
+            "Insufficient Balance of uAD token "
+        );
+
+        require(
+            IERC20(uADToken).allowance(msg.sender, address(this)) >=
+                _amountuAD * 10**18,
+            "You must approve uAD token before transfer"
```

Comment:
`uAD` has the ability to be renamed. I would suggest reading the token symbol from the smart contract.

Reply:
Okay, let me read the token symbol from the smart contract and will change the name as Dollar

Reply:
And after reading it I need to replace that in the require() statement